File: todoapp_flutter/lib/theme/app_theme.dart
import 'package:flutter/material.dart';

/// Brand colors shared across light/dark palettes.
class AppColors {
  static const Color brandBlue = Color(0xFF2563EB);
  static const Color brandRed = Color(0xFFDC2626);
  static const Color lightBg = Color(0xFFF3F4F6);
  static const Color lightSurface = Color(0xFFFFFFFF);
  static const Color lightSurfaceAlt = Color(0xFFF9FAFB);
  static const Color lightBorder = Color(0xFFE5E7EB);
  static const Color darkBg = Color(0xFF111827);
  static const Color darkSurface = Color(0xFF1F2937);
  static const Color darkSurfaceAlt = Color(0xFF374151);
  static const Color darkBorder = Color(0xFF4B5563);
}

class AppTheme {
  static ThemeData light() {
    final colorScheme = ColorScheme.light(
      primary: AppColors.brandBlue,
      onPrimary: Colors.white,
      secondary: AppColors.brandBlue,
      error: AppColors.brandRed,
      onError: Colors.white,
      surface: AppColors.lightSurface,
      onSurface: const Color(0xFF111827),
      surfaceContainerHighest: AppColors.lightSurfaceAlt,
      outline: AppColors.lightBorder,
    );

    return ThemeData(
      useMaterial3: true,
      brightness: Brightness.light,
      colorScheme: colorScheme,
      scaffoldBackgroundColor: AppColors.lightBg,
      appBarTheme: const AppBarTheme(
        backgroundColor: AppColors.lightSurface,
        foregroundColor: Color(0xFF111827),
        elevation: 0,
      ),
      elevatedButtonTheme: ElevatedButtonThemeData(
        style: ElevatedButton.styleFrom(
          backgroundColor: AppColors.brandBlue,
          foregroundColor: Colors.white,
        ),
      ),
      dividerColor: AppColors.lightBorder,
      cardColor: AppColors.lightSurface,
    );
  }

  static ThemeData dark() {
    final colorScheme = ColorScheme.dark(
      primary: AppColors.brandBlue,
      onPrimary: Colors.white,
      secondary: AppColors.brandBlue,
      error: AppColors.brandRed,
      onError: Colors.white,
      surface: AppColors.darkSurface,
      onSurface: const Color(0xFFF9FAFB),
      surfaceContainerHighest: AppColors.darkSurfaceAlt,
      outline: AppColors.darkBorder,
    );

    return ThemeData(
      useMaterial3: true,
      brightness: Brightness.dark,
      colorScheme: colorScheme,
      scaffoldBackgroundColor: AppColors.darkBg,
      appBarTheme: const AppBarTheme(
        backgroundColor: AppColors.darkSurface,
        foregroundColor: Color(0xFFF9FAFB),
        elevation: 0,
      ),
      elevatedButtonTheme: ElevatedButtonThemeData(
        style: ElevatedButton.styleFrom(
          backgroundColor: AppColors.brandBlue,
          foregroundColor: Colors.white,
        ),
      ),
      dividerColor: AppColors.darkBorder,
      cardColor: AppColors.darkSurface,
    );
  }
}

File: ToDoApp.KotlinMultiplatform/composeApp/src/commonMain/kotlin/com/example/todoappkotlinmultiplatform/ui/theme/AppTheme.kt
package com.example.todoappkotlinmultiplatform.ui.theme

import androidx.compose.material3.darkColorScheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.ui.graphics.Color

val BrandBlue = Color(0xFF2563EB)
val BrandRed = Color(0xFFDC2626)
val LightBg = Color(0xFFF3F4F6)
val LightSurface = Color(0xFFFFFFFF)
val LightSurfaceAlt = Color(0xFFF9FAFB)
val LightBorder = Color(0xFFE5E7EB)
val DarkBg = Color(0xFF111827)
val DarkSurface = Color(0xFF1F2937)
val DarkSurfaceAlt = Color(0xFF374151)
val DarkBorder = Color(0xFF4B5563)

val LightColorScheme = lightColorScheme(
    primary = BrandBlue,
    onPrimary = Color.White,
    secondary = BrandBlue,
    onSecondary = Color.White,
    error = BrandRed,
    onError = Color.White,
    background = LightBg,
    onBackground = Color(0xFF111827),
    surface = LightSurface,
    onSurface = Color(0xFF111827),
    surfaceVariant = LightSurfaceAlt,
    onSurfaceVariant = Color(0xFF374151),
    outline = LightBorder
)

val DarkColorScheme = darkColorScheme(
    primary = BrandBlue,
    onPrimary = Color.White,
    secondary = BrandBlue,
    onSecondary = Color.White,
    error = BrandRed,
    onError = Color.White,
    background = DarkBg,
    onBackground = Color(0xFFF9FAFB),
    surface = DarkSurface,
    onSurface = Color(0xFFF9FAFB),
    surfaceVariant = DarkSurfaceAlt,
    onSurfaceVariant = Color(0xFFD1D5DB),
    outline = DarkBorder
)

enum class AppThemeMode {
    LIGHT,
    DARK;

    fun toStorage(): String = when (this) {
        LIGHT -> "light"
        DARK -> "dark"
    }

    companion object {
        fun fromStorage(value: String?): AppThemeMode =
            if (value == "dark") DARK else LIGHT
    }
}
